libnavigation-core/src/test/java/com/mapbox/navigation/core/MapboxNavigationRerouteInterruptionTest.kt

```diff
@@ -0,0 +1,91 @@
+package com.mapbox.navigation.core
+
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.core.reroute.NavigationRerouteController
+import com.mapbox.navigation.core.reroute.RerouteController
+import com.mapbox.navigation.core.reroute.RerouteState
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verify
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.InternalCoroutinesApi
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.ParameterizedRobolectricTestRunner
+import org.robolectric.annotation.Config
+
+@ExperimentalPreviewMapboxNavigationAPI
+@Config(shadows = [ShadowReachabilityFactory::class])
+@InternalCoroutinesApi
+@ExperimentalCoroutinesApi
+@RunWith(ParameterizedRobolectricTestRunner::class)
+internal class MapboxNavigationRerouteInterruptionTest(
+    private val previousState: RerouteState?,
+    private val expectedNumberOfInterruptions: Int,
+) : MapboxNavigationBaseTest() {
+
+    companion object {
+
+        @JvmStatic
+        @ParameterizedRobolectricTestRunner.Parameters
+        fun data(): List<Array<Any?>> {
+            return listOf<Array<Any?>>(
+                arrayOf(null, 0),
+                arrayOf(RerouteState.Idle, 0),
+                arrayOf(RerouteState.Interrupted, 0),
+                arrayOf(RerouteState.FetchingRoute, 1),
+                arrayOf(mockk<RerouteState.Failed>(), 0),
+                arrayOf(mockk<RerouteState.RouteFetched>(), 0),
```

Comment:
Are you sure that we don't need to do anything in this case? I guess we do nothing because reroute can't happen when reroute controller in the fetched state, they all work on UI thread, and reroute controller switches immediately to Idle.
@RingerJK , @dzinad , what do you think about it?

Reply:
> reroute controller switches immediately to Idle

Clarification: it's our reroute controller immediately switches to idle. Who know what a custom reroute controller can do 🤔

Reply:
Well, yes, custom controllers could stay in RouteFetched state longer, our docs don't say they have to become idle asap. But there's not much we can do actually. We can only request a reroute, but if that's the case and the custom reroute controller is in RouteFetched state long time after the reroute finished, if would be an unnecessary reroute. So as I see it, it's fine as it is.

Reply:
code tests SDK implementation and expectation of one, so we should be good